sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java

```diff
@@ -553,6 +539,41 @@ private void onCaseChanged(Case existingCase, Case changedCase) {
 		}
 	}
 
+	private void assignOfficerOrSupervisorToTask(Case changedCase, Task task) {
+		if (changedCase.getSurveillanceOfficer() != null) {
+			task.setAssigneeUser(changedCase.getSurveillanceOfficer());
+		} else {
+			// TODO roles? what happens when there are no supervisors? assignee user cannot be null
```

Comment:
Please have a look at the task assignment process in the onCaseChanged method in the web backend and adjust the task assignment here accordingly, i.e. when there is no officer assigned to the case, use a random officer from the responsible district or, if there's none, from the place of stay district before falling back to supervisors.